test/sql/pktap.sql

```diff
@@ -136,7 +136,15 @@ SELECT * FROM check_test(
 );
 
 SELECT * FROM check_test(
-    col_is_pk( 'sometab', 'id', 'sometab.id should be a pk' ),
+       col_is_pk( 'public'::name, 'sometab'::name,'id'::name ),
+       true,
+       'col_is_pk( schema, table, column )',
+       'Column public.sometab(id) should be a primary key',
+       ''
+);
+
+SELECT * FROM check_test(
+    col_is_pk( 'sometab', 'id'::name, 'sometab.id should be a pk' ),
```

Comment:
Why is the name cast needed here? AFAICT requiring users to add that would break backwards compatibility, which I don't think we can do :(

Reply:
Unfortunately, https://travis-ci.org/theory/pgtap/jobs/614184912 shows that we have a problem when removing this cast. :(

One possible way to handle this would be C code, since we could then use one of the polymorphic pseudotypes. I'm not sure this feature is worth that though.

We could possibly have a config option to make to provide the old behavior.

Another possibility would be a version of the function that checked the first argument to see if it was a schema vs a table and acted accordingly (as well as throwing the correct error if neither existed). Before trying to code that we'd want to make sure we have full test cases for all the possible "does not exist" cases, both with and without a description provided. This would be unfortunately complex, but I think it's doable.

Well, or we could just tell people we're breaking backwards compatibility... but if we wanted to do that I think we'd need to do a 2.0.

Reply:
This is still an issue as well.